src/builder/value_parser.rs

```diff
@@ -160,8 +164,9 @@ impl ValueParser {
     /// assert_eq!(port, "80");
     /// ```
     pub const fn string() -> Self {
-        Self(ValueParserInner::String)
+        Self::STRING
     }
+    pub(crate) const STRING: Self = Self(ValueParserInner::String);
```

Comment:
This is unrelated to the specified motivation and these types of changes should at least be in a separate commit so I can more so tell what is pertinent for any given change.  However, I went ahead and did #3802